homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,163 @@
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "3"]
+            if CONF_REGIONS + "4" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "4"]
+            if CONF_REGIONS + "5" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "5"]
+            if CONF_REGIONS + "6" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "6"]
+
+            if len(config[CONF_REGIONS]) > 0:
+                tmp: dict[str, Any] = {}
+
+                for reg in config[CONF_REGIONS]:
+                    tmp[self._all_region_codes_sorted[reg]] = reg.split("_", 1)[0]
+
+                compact: dict[str, Any] = {}
+
+                for key, val in tmp.items():
+                    if val in compact:
+                        compact[val] = f"{compact[val]} + {key}"
+                        break
+                    compact[val] = key
+
+                config[CONF_REGIONS] = compact
+
+                return self.async_create_entry(title="NINA", data=config)
+
+            errors["base"] = "no_selection"
+
+        self.split_regions()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_REGIONS + "1"): cv.multi_select(self.regions_a),
+                    vol.Optional(CONF_REGIONS + "2"): cv.multi_select(self.regions_b),
+                    vol.Optional(CONF_REGIONS + "3"): cv.multi_select(self.regions_c),
+                    vol.Optional(CONF_REGIONS + "4"): cv.multi_select(self.regions_d),
+                    vol.Optional(CONF_REGIONS + "5"): cv.multi_select(self.regions_e),
+                    vol.Optional(CONF_REGIONS + "6"): cv.multi_select(self.regions_f),
+                    vol.Required(CONF_MESSAGE_SLOTS, default=5): vol.All(
+                        int, vol.Range(min=1, max=20)
+                    ),
+                    vol.Required(CONF_FILTER_CORONA, default=True): cv.boolean,
+                }
+            ),
+            errors=errors,
+        )
+
+    @staticmethod
+    def swap_key_value(dict_to_sort: dict[str, str]) -> dict[str, str]:
+        """Swap keys and values in dict."""
+        all_region_codes_swaped: dict[str, str] = {}
+
+        for key, value in dict_to_sort.items():
+            if value not in all_region_codes_swaped:
+                all_region_codes_swaped[value] = key
+            else:
+                for i in range(len(dict_to_sort)):
+                    tmp_value: str = value + "_" + str(i)
+                    if tmp_value not in all_region_codes_swaped:
+                        all_region_codes_swaped[tmp_value] = key
+                        break
+
+        return dict(sorted(all_region_codes_swaped.items(), key=lambda ele: ele[1]))
+
+    def split_regions(self) -> None:
+        """Split regions alphabetical."""
+        for i, name in self._all_region_codes_sorted.items():
```

Comment:
Can we give `i` a better name?

Reply:
Also make sure that we are not unnecessarily going through the whole loop if the region was already assigned